pkg/kvstore/kvstore.go

```diff
@@ -32,54 +34,59 @@ var (
 // This package implements a very basic key/value store.  It is up to the
 // caller to provision the namespace.
 
-type KeyValueStore struct {
-	b Backend
-
-	kv map[string][]byte
-
-	fileName string
-
-	l sync.RWMutex
+type KeyValueStore interface {
+	Set(ctx context.Context, key string, value []byte) error
+	Get(key string) ([]byte, error)
+	List(re string) (map[string][]byte, error)
+	Delete(ctx context.Context, key string) error
+	Save(ctx context.Context) error
+	Name() string
 }
 
-type Backend interface {
-	// Creates path and overwrites whatever is there.
-	Upload(ctx context.Context, r io.Reader, pth string) error
-
-	// Downloads from the given path.
-	Download(ctx context.Context, pth string) (io.ReadCloser, error)
+type kv struct {
+	b    Backend
+	kv   map[string][]byte
+	name string
+	l    sync.RWMutex
+}
 
-	// Moves the given path.
-	Mv(ctx context.Context, fromPath, toPath string) error
+func fileName(name string) string {
+	return fmt.Sprintf("%s.dat", name)
 }
 
 // Create a new KeyValueStore instance using the given Backend with the given
 // file.  If the file exists on the Backend, it is restored.
-func NewKeyValueStore(op trace.Operation, store Backend, fileName string) (*KeyValueStore, error) {
-	p := &KeyValueStore{
-		b:        store,
-		kv:       make(map[string][]byte),
-		fileName: fileName,
+func NewKeyValueStore(ctx context.Context, store Backend, name string) (KeyValueStore, error) {
```

Comment:
It's really up to you guys if you'd rather use the `Context` interface vs the `Operation` struct.  The interface is fine if all you're interested in is following the mutations of a context across API boundaries.  However, you'd need to be diligent in using the helper functions in the trace package and not the `context` package otherwise the chain is broken.   Also the `Operation` struct has some useful logging features which help you trace an operation by ID when using `Infof`, `Debugf`, and the like.

But for this use case, it's really up to you if you want to use them.  At some point @hickeng and I may go through and replace all instances of `context` with `Operation`.  It's already proven useful in debugging CDEs.